src/com/amazon/ion/impl/bin/IonManagedBinaryWriter.java

```diff
@@ -725,414 +124,279 @@ public boolean isFieldNameSet()
-    {
-        startLocalSymbolTableIfNeeded(true);
+    public void setFieldNameSymbol(SymbolToken token) {
+        currentWriter.setFieldNameSymbol(intern(token));
     }
 
-    public void setTypeAnnotations(final String... annotations)
-    {
-        if (annotations == null)
-        {
-            user.setTypeAnnotationSymbols((SymbolToken[]) null);
-        }
-        else
-        {
-            final SymbolToken[] tokens = new SymbolToken[annotations.length];
-            for (int i = 0; i < tokens.length; i++)
-            {
+    public void setTypeAnnotations(final String... annotations) {
+        if (annotations == null) {
+            currentWriter.setTypeAnnotationSymbols((SymbolToken[]) null);
+            //null is illegal in an annotation, why do we need this logic?
+        } else {
+            SymbolToken[] tokens = new SymbolToken[annotations.length];
+            for (int i = 0; i < tokens.length; i++) {
                 tokens[i] = intern(annotations[i]);
             }
-            user.setTypeAnnotationSymbols(tokens);
+            currentWriter.setTypeAnnotationSymbols(tokens);
         }
     }
 
     public void setTypeAnnotationSymbols(final SymbolToken... annotations)
     {
-        if (annotations == null)
-        {
-            user.setTypeAnnotationSymbols((SymbolToken[]) null);
-        }
-        else
-        {
-            for (int i = 0; i < annotations.length; i++)
-            {
-                annotations[i] = intern(annotations[i]);
+        if (annotations == null) {
+            currentWriter.setTypeAnnotationSymbols((SymbolToken[]) null);
+        } else {
+            for (int i = 0; i < annotations.length; i++) {
+                annotations[i] = intern(annotations[i]);//this is going to degrade perf badly TODO fix when import descriptors are added to ST
             }
-            user.setTypeAnnotationSymbols(annotations);
+            currentWriter.setTypeAnnotationSymbols(annotations);
         }
     }
 
-    public void addTypeAnnotation(final String annotation)
-    {
-        final SymbolToken token = intern(annotation);
-        user.addTypeAnnotationSymbol(token);
+    public void addTypeAnnotation(final String annotation) {
+        currentWriter.addTypeAnnotationSymbol(intern(annotation));
     }
 
-    // Container Manipulation
+    public void addTypeAnnotationSymbol(final SymbolToken annotation) {
+        currentWriter.addTypeAnnotationSymbol(intern(annotation));
+    }
 
+    // Container Manipulation
     public void stepIn(final IonType containerType) throws IOException
     {
-        userState.beforeStepIn(this, containerType);
-        user.stepIn(containerType);
+        if(currentWriter.getDepth() == 0 && user.hasTopLevelSymbolTableAnnotation() && containerType == STRUCT){//only true when the user writer wrote the lst st.
+            currentWriter = lstWriter;
+            user.clearAnnotations();//empties the prepped annotations on the user writer
```

Comment:
the annotation/s are already prepped on the user writer, we will not actually serialize an lst at this point unless the user calls flush after we absorb their lst context. We need to delete the annotations from the user writer because only the symbol writer should be serializing those system symbols and only after the user calls flush from the top level.